clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -1669,6 +1669,35 @@ protected boolean shouldRetain(RecordBatch recordBatch, Record record) {
     }
 
     @Test
+    public void testUpdatePositionOnEmptyBatch() {
```

Comment:
If `writeEmptyHeader` is only used by test code and is moved to a test utils class, then i guess we don't need this?